core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -525,14 +531,14 @@ private[spark] class MesosClusterScheduler(
         d.retryState.get.nextRetry.before(currentTime)
       }
 
-      scheduleTasks(
+      currentOffers = scheduleTasks(
```

Comment:
similarly I don't get the point of assigning this var every time you call `scheduleTasks`. It's the same thing, isn't it?